bed/methods.go

```diff
@@ -14,17 +14,18 @@ func (b Bed) GetChrom() string {
 }
 
 func (b Bed) GetChromStart() int {
-	return int(b.ChromStart)
+	return b.ChromStart
 }
 
 func (b Bed) GetChromEnd() int {
-	return int(b.ChromEnd)
+	return b.ChromEnd
 }
 
-func (b Bed) UpdateLift(c string, start int, end int) {
+func (b Bed) UpdateCoord(c string, start int, end int) interface{} {
```

Comment:
This is totally fine, though I imagine the empty interface return can get a bit unwieldy. This is not a necessary change, but you might consider instead something like `func (b *Bed) SetCoord(...)`. That way you don't need a return value. I'm not positive, but I think pointer here would go on the stack, not the heap, so the performance would be pretty much identical between the two options.

Reply:
i was going to ask you about this, because i tried that first and then it said Bed could not implement Lift because it seemed like it couldn't recognize the function (b *Bed) as taking in a Bed. How did you get around this?

Reply:
That is a good point. I did not think of that when I commented. In looking a bit deeper it seems that the general consensus is to use pointers for all methods in a set if any of them require pointers. This may be be worth a larger conversation. I'll copy below an excerpt from the FAQ on go.dev

Should I define methods on values or pointers?
func (s *MyStruct) pointerMethod() { } // method on pointer
func (s MyStruct)  valueMethod()   { } // method on value
For programmers unaccustomed to pointers, the distinction between these two examples can be confusing, but the situation is actually very simple. When defining a method on a type, the receiver (s in the above examples) behaves exactly as if it were an argument to the method. Whether to define the receiver as a value or as a pointer is the same question, then, as whether a function argument should be a value or a pointer. There are several considerations.

First, and most important, does the method need to modify the receiver? If it does, the receiver must be a pointer. (Slices and maps act as references, so their story is a little more subtle, but for instance to change the length of a slice in a method the receiver must still be a pointer.) In the examples above, if pointerMethod modifies the fields of s, the caller will see those changes, but valueMethod is called with a copy of the caller's argument (that's the definition of passing a value), so changes it makes will be invisible to the caller.

By the way, in Java method receivers are always pointers, although their pointer nature is somewhat disguised (and there is a proposal to add value receivers to the language). It is the value receivers in Go that are unusual.

Second is the consideration of efficiency. If the receiver is large, a big struct for instance, it will be much cheaper to use a pointer receiver.

Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used. See the section on [method sets](https://go.dev/doc/faq#different_method_sets) for details.

For types such as basic types, slices, and small structs, a value receiver is very cheap so unless the semantics of the method requires a pointer, a value receiver is efficient and clear.

Reply:
yes certainly looks like a longer conversation. I'm open to switching to pointers, though it would be quite a rewrite for interval and other packages